### PR TITLE
JBIDE-28177: Switch to odo 2.4.2

### PR DESCRIPTION
**Pull request by jeffmaury**
Signed-off-by: Jeff MAURY <jmaury@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/CreateComponentWizardPage.java [hunk 1]
```diff
@@ -155,54 +132,31 @@ protected void doCreateControls(Composite parent, DataBindingContext dbc) {
 		Label componentTypesLabel = new Label(parent, SWT.NONE);
 		componentTypesLabel.setText("Component type:");
 		GridDataFactory.fillDefaults().align(SWT.FILL, SWT.TOP).applyTo(componentTypesLabel);
-		Tree componentTypesTree = new Tree(parent, SWT.SINGLE | SWT.BORDER);
+		org.eclipse.swt.widgets.List componentTypesList = new org.eclipse.swt.widgets.List(parent, SWT.SINGLE | SWT.BORDER);
```
Comment (sbouchet) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/CreateComponentWizardPage.java:135]: When creating a component, i do not see what type is selected by alizer in the list (used quarkus). there is no scrolling neither .

**Review by sbouchet — COMMENTED** [review]
Also, component UI creation dialog does not resize the list when expanding

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/CreateComponentWizardPage.java (hunk 1 above)
Comment (jeffmaury) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/CreateComponentWizardPage.java:135]: Will fix the scrollbar issue but can't reproduce the default selection from alizer (tested with a Quarkus and SpringBoot projects). Make sure your are using registry.devfile.io in your settings

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/odo/OdoCli.java [hunk 2]
```diff
@@ -912,9 +912,9 @@ public void link(String project, String application, String component, String co
     UsageStats.getInstance().link();
     try {
       if (port != null) {
-        execute(new File(context), command, envVars, "link", source, "--port", port.toString(), "--wait");
```
Comment (sbouchet) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/odo/OdoCli.java:918]: linking component with port fails 
Capture d’écran de 2021-11-25 12-32-30
:

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/applicationexplorer/LinkServiceHandler.java [hunk 3]
```diff
@@ -59,7 +59,7 @@ private void execute(Shell shell, LinkModel<Service> model, ComponentElement com
 		        "Linking component " + model.getComponentName() + " to service " + model.getTarget());
 		try {
 			model.getOdo().link(model.getProjectName(), model.getApplicationName(), component.getWrapped().getName(),
-			        component.getWrapped().getPath(), model.getTarget().getKind() + '/' +model.getTarget().getName(), null);
+			        component.getWrapped().getPath(), model.getTarget().getKind() + '/' +model.getTarget().getName());
 			LabelNotification.openNotification(notification, shell,
 			        "Component " + model.getComponentName() + " linked to service " + model.getTarget());
```
Comment (sbouchet) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/applicationexplorer/LinkServiceHandler.java:64]: notification message displays Service.tostring() , should be Service.getName()

**Review by sbouchet — APPROVED** [review]
LGTM.

**Pull request merged**